File: AddSwaggerTags.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.util.regex.*;

public class AddSwaggerTags {

    private static final Map<String, String[]> MAPPINGS = new HashMap<>();

    static {
        MAPPINGS.put("AuthController", new String[]{"Identity APIs", "Kuldeep Pachori"});
        MAPPINGS.put("AppUserController", new String[]{"Identity APIs", "Kuldeep Pachori"});
        MAPPINGS.put("CurrencyController", new String[]{"Reference Data APIs", "Kuldeep Pachori"});
        MAPPINGS.put("CountryController", new String[]{"Reference Data APIs", "Kuldeep Pachori"});
        MAPPINGS.put("MarketController", new String[]{"Reference Data APIs", "Kuldeep Pachori"});
        MAPPINGS.put("CustomerController", new String[]{"Customer APIs", "Kuldeep Pachori"});
        MAPPINGS.put("AccountController", new String[]{"Account APIs", "Kuldeep Pachori"});

        MAPPINGS.put("FundController", new String[]{"Funds APIs", "Sanket Ganje"});
        MAPPINGS.put("FxRateController", new String[]{"FX APIs", "Sanket Ganje"});
        MAPPINGS.put("FundPriceController", new String[]{"Pricing APIs", "Sanket Ganje"});
        MAPPINGS.put("ImportJobController", new String[]{"Import APIs", "Sanket Ganje"});
        MAPPINGS.put("ImportBatchController", new String[]{"Import APIs", "Sanket Ganje"});
        MAPPINGS.put("ImportItemController", new String[]{"Import APIs", "Sanket Ganje"});

        MAPPINGS.put("PortfolioController", new String[]{"Portfolio APIs", "Rushikesh Mind"});
        MAPPINGS.put("TradeOrderController", new String[]{"Trading APIs", "Rushikesh Mind"});
        MAPPINGS.put("TradeExecutionController", new String[]{"Trading APIs", "Rushikesh Mind"});
        MAPPINGS.put("SettlementController", new String[]{"Settlement APIs", "Rushikesh Mind"});
        MAPPINGS.put("JournalController", new String[]{"Ledger APIs", "Rushikesh Mind"});
        MAPPINGS.put("LedgerAccountController", new String[]{"Ledger APIs", "Rushikesh Mind"});

        MAPPINGS.put("DividendEventController", new String[]{"Dividends APIs", "Mayur Mali"});
        MAPPINGS.put("DividendAllocationController", new String[]{"Dividends APIs", "Mayur Mali"});
        MAPPINGS.put("ReconciliationRunController", new String[]{"Reconciliation APIs", "Mayur Mali"});
        MAPPINGS.put("ReconciliationItemController", new String[]{"Reconciliation APIs", "Mayur Mali"});
        MAPPINGS.put("ReconciliationResolutionController", new String[]{"Reconciliation APIs", "Mayur Mali"});
        MAPPINGS.put("AuditEventController", new String[]{"Audit APIs", "Mayur Mali"});
    }

    public static void main(String[] args) throws IOException {
        Path startPath = Paths.get("e:/wealth-link-platform/src/main/java/com/wealthlink");
        
        Files.walk(startPath)
            .filter(path -> path.toString().endsWith("Controller.java"))
            .forEach(path -> {
                try {
                    processFile(path);
                } catch (IOException e) {
                    e.printStackTrace();
                }
            });
        System.out.println("Done processing controllers.");
    }

    private static void processFile(Path path) throws IOException {
        String fileName = path.getFileName().toString();
        String className = fileName.replace(".java", "");

        if (!MAPPINGS.containsKey(className)) {
            return;
        }

        String[] tagInfo = MAPPINGS.get(className);
        String domain = tagInfo[0];
        String dev = tagInfo[1];

        String content = new String(Files.readAllBytes(path));

        // Clean up previous tags if present
        content = content.replaceAll("@Tag\\(.*?\\)\\n", "");

        if (!content.contains("import io.swagger.v3.oas.annotations.tags.Tag;")) {
            content = content.replace("import org.springframework.web.bind.annotation.*;", 
                    "import org.springframework.web.bind.annotation.*;\nimport io.swagger.v3.oas.annotations.tags.Tag;");
        }

        String tagAnnotation = String.format("@Tag(name = \"%s\", description = \"Maintained by: %s\")\n@RestController", domain, dev);
        content = content.replace("@RestController", tagAnnotation);

        Files.write(path, content.getBytes());
    }
}

File: add_swagger_tags.py
import os
import re

DEV1 = "Kuldeep Pachori"
DEV2 = "Sanket Ganje"
DEV3 = "Rushikesh Mind"
DEV4 = "Mayur Mali"

MAPPINGS = {
    "AuthController": ("Identity", DEV1),
    "AppUserController": ("Identity", DEV1),
    "CurrencyController": ("Reference Data", DEV1),
    "CountryController": ("Reference Data", DEV1),
    "MarketController": ("Reference Data", DEV1),
    "CustomerController": ("Customer", DEV1),
    "AccountController": ("Account", DEV1),
    
    "FundController": ("Funds", DEV2),
    "FxRateController": ("FX", DEV2),
    "FundPriceController": ("Pricing", DEV2),
    "ImportJobController": ("Import", DEV2),
    "ImportBatchController": ("Import", DEV2),
    "ImportItemController": ("Import", DEV2),
    
    "PortfolioController": ("Portfolio", DEV3),
    "TradeOrderController": ("Trading", DEV3),
    "TradeExecutionController": ("Trading", DEV3),
    "SettlementController": ("Settlement", DEV3),
    "JournalController": ("Ledger", DEV3),
    "LedgerAccountController": ("Ledger", DEV3),
    
    "DividendEventController": ("Dividends", DEV4),
    "DividendAllocationController": ("Dividends", DEV4),
    "ReconciliationRunController": ("Reconciliation", DEV4),
    "ReconciliationItemController": ("Reconciliation", DEV4),
    "ReconciliationResolutionController": ("Reconciliation", DEV4),
    "AuditEventController": ("Audit", DEV4),
}

base_path = "e:/wealth-link-platform/src/main/java/com/wealthlink"

for root, _, files in os.walk(base_path):
    for file in files:
        if file.endswith("Controller.java"):
            file_path = os.path.join(root, file)
            class_name = file.replace(".java", "")
            
            if class_name in MAPPINGS:
                domain, dev = MAPPINGS[class_name]
                
                with open(file_path, "r", encoding="utf-8") as f:
                    content = f.read()
                    
                if "io.swagger.v3.oas.annotations.tags.Tag" in content:
                    continue
                    
                # Add import
                content = content.replace("import org.springframework.web.bind.annotation.*;", 
                                          "import org.springframework.web.bind.annotation.*;\nimport io.swagger.v3.oas.annotations.tags.Tag;")
                
                # Add tag
                tag_annotation = f'@Tag(name = "{domain} APIs", description = "Maintained by: {dev}")\n@RestController'
                content = content.replace("@RestController", tag_annotation)
                
                with open(file_path, "w", encoding="utf-8") as f:
                    f.write(content)
                    
print("Done")
